azure_functions_worker/utils/common.py

```
@@ -146,11 +146,26 @@ def __init__(self, file_name: str) -> None:
             f'Invalid file name: {file_name}')
 
 
-def validate_script_file_name(file_name: str):
-    # First character can be a letter, number, or underscore
-    # Following characters can be a letter, number, underscore, hyphen, or dash
-    # Ending must be .py
+def validate_script_file(function_path: Union[str, os.PathLike],
+                         file_name: str) -> (bool, bool):
+    """ Validate the script file name.
+    Returns:
+    T, T : V2 programming model
+    T, F : V2 programming model but with invalid file name
+    F, T : File path does not exist but filename is valid.
+           Case 1: V1 programming model
+           Case 2: V2 programming model but with invalid file path
+                If V2, return T, F
+    F, F : Invalid function
+    """
     pattern = re.compile(r'^[a-zA-Z0-9_][a-zA-Z0-9_\-]*\.py$')
-    if not pattern.match(file_name):
-        raise InvalidFileNameError(file_name)
-    return True
+    path_exists = os.path.exists(function_path)
+    valid_file_name = pattern.match(file_name)
+    app_setting_set = PYTHON_SCRIPT_FILE_NAME in os.environ.copy()
+
+    # Handles special case where App Setting is set to incorrect file name
+    # for V2 programming model
+    if app_setting_set and not path_exists and valid_file_name:
+        return True, False
```

move this check at top to cut overhead when failing fast